Openstack baremetal compute nodes

pierreprinetti opened the pull request.

pierreprinetti committed:
wip

racedo commented on enhancements/openstack-baremetal-compute.md:
```diff
@@ -0,0 +1,170 @@
+enhancement will ensure that OpenShift installations can use them as Compute
+nodes.
+
+### Goals
```
A goal is to deploy workers on BMs and masters on VMs too. This is the main request from customers for this use case

pierreprinetti replied:
Thank you! I have changed the wording in the hopes it's clearer.

pierreprinetti committed:
Use consistent names to tell apart OSP/OCP nodes

OSP Nova instance --> machine

OCP node --> either Compute Node or Control Plane node

pierreprinetti committed:
Remove template introduction

mandre commented on enhancements/openstack-baremetal-compute.md:
```diff
@@ -0,0 +1,140 @@
+be used as OpenShift Compute Nodes, when they are made available as Nova
+flavors.
+
+In this enhancement, OpenShift's Control Plane nodes run on regular Nova
```
Unless there anything preventing us from running control plane on BM nodes, I believe we shouldn't explicitly make it out of scope.
We should however make it a requirement that this enhancement allows running a mix of BM and VM nodes, and document a use case where a customer wants to run the Control Plane on BM nodes for increased performance since this is the most typical use case for the feature.

pierreprinetti replied:
I like your technical translation of the [Product requirements](https://jira.coreos.com/browse/OSASINFRA-992)

mandre commented on enhancements/openstack-baremetal-compute.md:
```diff
@@ -0,0 +1,140 @@
+* Baremetal machines provisioned by Nova through Ironic can be used as
+  OpenShift Compute Nodes, while the Control Plane nodes run on virtualised
+  hardware.
+* Baremetal machines can be attached and detached from a running OpenShift
```
I find the formulation a little bit confusing. How about: `Baremetal machines can be added or removed from a running OpenShift cluster...` ?

mandre commented on enhancements/openstack-baremetal-compute.md:
```diff
@@ -0,0 +1,140 @@
+
+TODO: alternatives
+	- OVB on RDO?
+        - MOC is planning on adding Ironic
```
As I currently understand it, we'll need an OVB VM running in our CI cloud that serves the BMC for the virtual BM nodes.
We'll also need to implement a solution to dynamically add / remove nodes from vbmc running on the VM.

pierreprinetti committed:
Document baremetal workers as a user story

while leaving room for mix-and-matching in the Summary.

pierreprinetti commented:
OK closing this and opening a WIP against the enhancements repo. Thank you!
